### PR TITLE
Let high priority notifications be sent if there was already one sent…

### PR DESCRIPTION
**Pull request by rhumaric**
… and the priority changed

Instead of looking if there was a previous notification, the check looks at whether the `action_priority` changed. This ensures notifications are sent appropriately if a task goes down to `medium` or `low` priority before switching back to `high`.

### CODE REVIEW

**Comment by rhumaric** [comment]
- An extra notification has been added to let Volunteer know when they've been unassigned from an action
- Changed when the notification for being assigned to an action get sent, so it gets sent properly when being assigned, then unassigned and finally re-assigned to an action
- Only send the action for not being assigned if you had not received any email regarding assignment before, preventing volunteers to receive it after they were assigned.

File: api/notifications/notifications.py [hunk 1]
```diff
@@ -54,18 +55,23 @@ def create_action_notifications(action):
...
+
+        # Use `in` as the value might be `None`, in case of first assignment
+        if 'assigned_volunteer_id' in changed:
             create([action.assigned_volunteer.email], action=action,
                    notification_type=NotificationTypes.VOLUNTEER_ASSIGNED)
 
-        # 2. Let those not assigned know.
-        # Only send a notification to volunteers that have not received this email for this action.
+        # 2. Let the previously assigned volunteer know
+            if (changed.get('assigned_volunteer_id')):
```
Comment (cgillions) [inline api/notifications/notifications.py:65]: We've already checked `if 'assigned_volunteer_id' in changed`, so we shouldn't need to check again.

File: api/notifications/signals.py [hunk 2]
```diff
@@ -24,16 +24,21 @@ def post_save_user(sender, instance, created, **kwargs):
...
     # the foreign key on the notifications table
     transaction.on_commit(
         lambda: django_rq.enqueue(
-            notifications.create_action_notifications, instance, result_ttl=0
+            notifications.create_action_notifications, instance, changed, result_ttl=0
         ), using=using)
 
 
-@receiver(post_save, sender=Notification, dispatch_uid="NotificationSave")
+@ receiver(post_save, sender=Notification, dispatch_uid="NotificationSave")
```
Comment (cgillions) [inline api/notifications/signals.py:41]: I assume this space is accidental?

File: api/notifications/templates/notifications/volunteer_unassigned_message.txt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Hi there,
+ 
+This action has now been reassigned to another volunteer. Thanks for your support with it:
+{{ action_url }}
+ 
+If there is any information you'd like to pass to the new volunteer, please email: info@kwalliance.co.uk.
```
Comment (cgillions) [inline api/notifications/templates/notifications/volunteer_unassigned_message.txt:6]: Can we use the coordinator email variable, instead of hardcoding?

Comment (cgillions) [reply api/notifications/templates/notifications/volunteer_unassigned_message.txt:6]: Done

File: api/notifications/signals.py (hunk 2 above)
Comment (cgillions) [reply api/notifications/signals.py:41]: Removed

File: api/notifications/notifications.py (hunk 1 above)
Comment (cgillions) [reply api/notifications/notifications.py:65]: Removed repeated check

Comment (cgillions) [reply api/notifications/notifications.py:65]: Undone - this _was_ needed.

Comment (rhumaric) [reply api/notifications/notifications.py:65]: My bad, I should have put a comment. The first one checked if there was a change of assigned volunteer, the second if there was actually a previous volunteer to notify.

File: api/notifications/signals.py (hunk 2 above)
Comment (rhumaric) [reply api/notifications/signals.py]: ooh, yeah, hadn't noticed it, good catch :)

**Pull request merged**